api/models/graphite.go

```diff
@@ -15,7 +15,8 @@ type GraphiteRender struct {
 	From          string   `json:"from" form:"from"`
 	Until         string   `json:"until" form:"until"`
 	To            string   `json:"to" form:"to"`
-	Format        string   `json:"format" form:"format" binding:"In(,json,msgp)"`
+	Format        string   `json:"format" form:"format" binding:"In(,json,msgp,pickle)"`
+	LocalOnly     bool     `json:"local" form:"local"` //this is set to true by graphite-web when it passes request to cluster servers
```

Comment:
graphite should only send queries with all functions stripped from the target param. (ie, it just sends `target=<pathExpr>&local=true`)

> "local" is ambiguous here.

Nothing we can do about that.  "local" is the query param that graphite sends.

The only reason we handle this query param is to prevent an infinite loop if either graphite or metrictank incorrectly parse the target.  if graphite sends a target with a pathExpr wrapped in a function, or MT thinks the target cant be processed locally and needs to be proxied, bad things will happen.

We can change the stuct field to "MetrictankOnly" if that helps.